On Ubuntu, the "npm start" command won't compile without "sudo",

fmanea opened the pull request:
So for users using Ubuntu, it would be more user friendly to know what to do to make it work
from the first command, and not seak for help.

Also, an incomplete path to index.jsx appears on the demo page

It shows index.js, instead of index.jsx.

Signed-off-by: Flavian-Costin Manea <maneaflavian@gmail.com>

<!--- Provide a general summary of your changes in the Title above -->

<!-- Thanks for submitting a pull request! -->
<!-- Please provide enough information so that others can review your pull request -->
<!-- For more information, see the `CONTRIBUTING` guide. -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- Describe the current and new situation/behavior --->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Chore (tests, refactors, and fixes)
- [ ] New feature (adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] I have read the **CONTRIBUTING** document.

## Closing issues
<!-- Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such). -->
Fixes #

fmanea committed:
On Ubuntu, the "npm start" command won't compile without "sudo",

So for users using Ubuntu, it would be more user friendly to know what to do to make it work
from the first command, and not seak for help.

Signed-off-by: Flavian-Costin Manea <maneaflavian@gmail.com>

fmanea committed:
The file extension for index.js  is .jsx

Signed-off-by: Flavian-Costin Manea <maneaflavian@gmail.com>

mhuggins commented on lib/steps/copyFiles.js:
```diff
@@ -51,7 +51,7 @@ function () {
             console.log(_chalk.default.white("To run your dApp, execute:"));
             console.log(_chalk.default.white.bold("1. 'cd ".concat(name, "'")));
-            console.log(_chalk.default.white.bold("2. 'npm start' or 'yarn start'"));
+            console.log(_chalk.default.white.bold("2. 'npm start' or 'yarn start' (use sudo if working on Ubuntu)"));
```
This is likely due to an issue with how you installed npm or some other associated CLI program.  /cc @DalderupMaurice @jeroenptrs to confirm, but I believe that needing sudo here is a sign that you have something on your system installed/configured incorrectly.

DalderupMaurice replied:
Normally when you create a project with the create-nos-dapp tool, the "npm install" or "yarn install" is done for you. Only thing left is to cd into the folder and start it. 
Could you confirm it fails to automatically install the dependencies? 

Also, I've encountered the problem occasionally with having to do `sudo xxx install` aswell on AWS servers (aka ubuntu) - but I am pretty sure it has something to do with npm not being installed correctly. I'm not sure what exactly goes wrong in these cases.

fmanea replied:
Hello, I also encountered the same problem on Windows 10, if powershell wasn't started with administrative rights "npm start" would fail in the same way, as it did on Ubuntu.

fmanea replied:
Ubuntu environment: Ubuntu 18.04 LTS
Commands that have been run:
```sudo apt-get install npm```
```sudo npm i -g @nosplatform/create-nos-dapp``` (it wouldn't work without sudo, because it wouldn't have the rights to copy inside  /usr/local/lib/node_modules)
```create-nos-dapp```
```sudo npm start``` ( it wouldn't build without sudo)

Maybe the problem may be that the npm package was installed globally, I haven't try installing the package locally.

jeroenptrs replied:
Can you remove the (use sudo if working on Ubuntu) mentions? here and in `src/steps/copyFiles.js`

fmanea replied:
I have reverted the commit.

mhuggins commented on packages/react/src/views/__tests__/__snapshots__/App.spec.jsx.snap:
```diff
@@ -27,7 +27,7 @@ exports[`Components Header should render 1`] = `
     To get started, edit 
     <code>
-      src/views/App/index.js
+      src/views/App/index.jsx
```
👍

mhuggins reviewed (commented):
I'll wait for the other guys to confirm, but I think the only change we would probably want to include here is the change from `js` => `jsx`.

jeroenptrs commented:
@fmanea are you using nvm on linux/windows? https://docs.npmjs.com/getting-started/installing-node#using-a-version-manager-to-install-nodejs-and-npm

fmanea commented:
After reading your link, it seemed that using nvm could solve the issue, I installed npm using 
```sudo apt-get install npm```
Well, I guess that only the typo can be merged, should I do a separate pull request for it?
Or maybe update the README.md to suggest using nvm.
Best regards,
Flavian

jeroenptrs commented:
@fmanea yeah you can use this pull to suggest using nvm!

fmanea committed:
Suggest using nvm when install npm to prevent permision issues.

There are problems on Windows and Linux when trying to use: npm start
If npm was installed via apt-get on Ubuntu. When this was the case, it
required administrative permisions to successfully compile the project.

My change was added in order to have a flawless use of create-nos-dapp
application.

Signed-off-by: Flavian-Costin Manea <maneaflavian@gmail.com>

fmanea commented:
@jeroenptrs I have added a new commit with the README.md change.

jeroenptrs commented on README.md:
```diff
@@ -29,8 +29,9 @@ The goal of this project is to provide a quickstart for creating nOS dApps with
 
 
 ## Installation
+Before we start, we need to install npm and in order to avoid permission issues on different OS, we suggest using nvm. You can follow the install instructions [here](https://docs.npmjs.com/getting-started/installing-node#using-a-version-manager-to-install-nodejs-and-npm).
```
Before we start, we want to make sure npm is installed via a version manager. To avoid permission issues on different OS, we suggest using nvm.

fmanea replied:
Done

fmanea committed:
Changes after code review

Signed-off-by: Flavian-Costin Manea <maneaflavian@gmail.com>

fmanea committed:
Revert "On Ubuntu, the "npm start" command won't compile without "sudo","

Changes were made in README.md
This reverts commit 3f9f9e765c49ea0675bb40f7bc19c3f094459819.

fmanea commented:
I've made the required changes.
